flask/app.py

```diff
@@ -1029,6 +1029,27 @@ def index():
         rule = self.url_rule_class(rule, methods=methods, **options)
         rule.provide_automatic_options = provide_automatic_options
 
+        for prior_rule in self.url_map.iter_rules():
```

Comment:
This is not very efficient - you iterate over all existing elements whenever you add a new one. So you end up with `sum(range(num_rules))` total iterations.